framework/src/vectorpostprocessors/VectorPostprocessorData.C

```diff
@@ -58,9 +58,24 @@ VectorPostprocessorData::getVectorPostprocessorHelper(const VectorPostprocessorN
                                                       const std::string & vector_name,
                                                       bool get_current)
 {
-  // Intentional use of RHS brackets on a std::map to do a multilevel retrieve or insert
-  auto & vec_struct = _values[vpp_name][vector_name];
+  // Intentional use of RHS brackets on a std::map to do a retrieve or insert
+  auto & vec_storage = _values[vpp_name];
 
+  // lambda for doing compairison on name (i.e., first item in pair)
+  auto comp = [&vector_name](std::pair<std::string, VectorPostprocessorState> & pair) {
+    return pair.first == vector_name;
+  };
+
+  // Search for the vector, if it is not located create the entry in the storage
+  auto iter = std::find_if(vec_storage.begin(), vec_storage.end(), comp);
+  if (iter == vec_storage.end())
+  {
+    vec_storage.emplace_back(
+        std::pair<std::string, VectorPostprocessorState>(vector_name, VectorPostprocessorState()));
+    iter = vec_storage.end() - 1; // can't use rbegin() because we need a forward iterator
```

Comment:
I was hoping for this option as well, but it is C++17

### Return value
```
(none) 	(until C++17)
A reference to the inserted element. 	(since C++17)
```